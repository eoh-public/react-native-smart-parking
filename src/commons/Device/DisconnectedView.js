import React, { memo } from 'react';
import { StyleSheet, View } from 'react-native';
import { IconOutline } from '@ant-design/icons-react-native';
import t from 'i18n';

import { Colors } from '../../configs';
import { CircleView } from '../../commons';
import Text from '../../commons/Text';
import DoorIcon from '../../../assets/images/Device/door-inactive.svg';
import BarrierIcon from '../../../assets/images/Device/barrier-inactive.svg';
import SensorIcon from '../../../assets/images/Device/sensor-inactive.svg';
import WifiOffIcon from '../../../assets/images/Device/wifi-off.svg';

const DeviceIcon = memo(({ sensor }) => {
  const { icon } = sensor;
  if (icon === 'door') {
    return <DoorIcon height={38} width={38} />;
  }
  if (icon === 'sensor') {
    return <SensorIcon height={38} width={38} />;
  }
  if (icon === 'barrier') {
    return <BarrierIcon height={38} width={38} />;
  }
  if (icon === 'wind') {
    return <SensorIcon height={38} width={38} />;
  }
  return <DoorIcon height={38} width={38} />;
});

const DisconnectedView = memo(({ sensor }) => (
  <View style={styles.statusContainer}>
    <CircleView size={80} backgroundColor={Colors.Gray3} style={styles.center}>
      <CircleView size={64} backgroundColor={Colors.BG} style={styles.center}>
        <DeviceIcon sensor={sensor} />
      </CircleView>
    </CircleView>
    <View style={styles.connectStatus}>
      <WifiOffIcon width={16} height={16} />
      <Text style={styles.redStatus}>{t('disconnected')}</Text>
    </View>
    <View style={styles.disconnectSuggestionContainer}>
      <View style={styles.row}>
        <IconOutline name={'alert'} size={16} />
        <Text bold style={styles.boldText}>
          {t('suggestions')}:
        </Text>
      </View>
      <View style={styles.infoRow}>
        <CircleView size={6} backgroundColor={Colors.Gray8} />
        <Text style={styles.noticeText}>{t('check_the_power')}</Text>
      </View>
      <View style={styles.infoRow}>
        <CircleView size={6} backgroundColor={Colors.Gray8} />
        <Text style={styles.noticeText}>{t('check_the_wifi')}</Text>
      </View>
    </View>
  </View>
));

const styles = StyleSheet.create({
  statusContainer: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: 16,
  },
  connectStatus: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 8,
    marginBottom: 24,
  },
  redStatus: {
    marginLeft: 10,
    fontSize: 14,
    color: Colors.Red6,
  },
  disconnectSuggestionContainer: {
    width: '100%',
    padding: 10,
  },
  row: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 9,
  },
  infoRow: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 16,
  },
  boldText: {
    fontSize: 12,
    marginLeft: 10,
    color: Colors.Gray8,
  },
  noticeText: {
    fontSize: 14,
    marginLeft: 8,
    color: Colors.Gray8,
  },
  center: {
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export { DisconnectedView };
