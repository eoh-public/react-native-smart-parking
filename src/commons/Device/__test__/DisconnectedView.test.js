import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { DisconnectedView } from '../DisconnectedView';

describe('Test DisconnectedView', () => {
  let tree;

  test('render DisconnectedView icon dooor', () => {
    const sensor = { icon: 'door' };
    act(() => {
      tree = renderer.create(<DisconnectedView sensor={sensor} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render DisconnectedView icon sensor', () => {
    const sensor = { icon: 'sensor' };
    act(() => {
      tree = renderer.create(<DisconnectedView sensor={sensor} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render DisconnectedView icon barrier', () => {
    const sensor = { icon: 'barrier' };
    act(() => {
      tree = renderer.create(<DisconnectedView sensor={sensor} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render DisconnectedView icon wind', () => {
    const sensor = { icon: 'wind' };
    act(() => {
      tree = renderer.create(<DisconnectedView sensor={sensor} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render DisconnectedView icon test', () => {
    const sensor = { icon: 'test' };
    act(() => {
      tree = renderer.create(<DisconnectedView sensor={sensor} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
