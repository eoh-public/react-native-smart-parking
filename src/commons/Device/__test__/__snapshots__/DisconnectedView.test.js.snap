// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DisconnectedView render DisconnectedView icon barrier 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5F5",
          "borderColor": "white",
          "borderRadius": 40,
          "borderWidth": 0,
          "height": 80,
          "width": 80,
        },
        undefined,
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F0F2F5",
            "borderColor": "white",
            "borderRadius": 32,
            "borderWidth": 0,
            "height": 64,
            "width": 64,
          },
          undefined,
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <SvgMock
        height={38}
        width={38}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 24,
        "marginTop": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "color": "#F5222D",
            "fontSize": 14,
            "marginLeft": 10,
          },
        ]
      }
    >
      Ngắt kết nối
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 9,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "antoutline",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 12,
              "marginLeft": 10,
            },
          ]
        }
      >
        Gợi ý
        :
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#595959",
              "borderColor": "white",
              "borderRadius": 3,
              "borderWidth": 0,
              "height": 6,
              "width": 6,
            },
            undefined,
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 14,
              "marginLeft": 8,
            },
          ]
        }
      >
        Kiểm tra lại nguồn điện của thiết bị.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#595959",
              "borderColor": "white",
              "borderRadius": 3,
              "borderWidth": 0,
              "height": 6,
              "width": 6,
            },
            undefined,
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 14,
              "marginLeft": 8,
            },
          ]
        }
      >
        Kiểm tra lại Wi-Fi/tín hiệu di động hoặc thử khởi động lại.
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test DisconnectedView render DisconnectedView icon dooor 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5F5",
          "borderColor": "white",
          "borderRadius": 40,
          "borderWidth": 0,
          "height": 80,
          "width": 80,
        },
        undefined,
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F0F2F5",
            "borderColor": "white",
            "borderRadius": 32,
            "borderWidth": 0,
            "height": 64,
            "width": 64,
          },
          undefined,
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <SvgMock
        height={38}
        width={38}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 24,
        "marginTop": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "color": "#F5222D",
            "fontSize": 14,
            "marginLeft": 10,
          },
        ]
      }
    >
      Ngắt kết nối
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 9,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "antoutline",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 12,
              "marginLeft": 10,
            },
          ]
        }
      >
        Gợi ý
        :
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#595959",
              "borderColor": "white",
              "borderRadius": 3,
              "borderWidth": 0,
              "height": 6,
              "width": 6,
            },
            undefined,
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 14,
              "marginLeft": 8,
            },
          ]
        }
      >
        Kiểm tra lại nguồn điện của thiết bị.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#595959",
              "borderColor": "white",
              "borderRadius": 3,
              "borderWidth": 0,
              "height": 6,
              "width": 6,
            },
            undefined,
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 14,
              "marginLeft": 8,
            },
          ]
        }
      >
        Kiểm tra lại Wi-Fi/tín hiệu di động hoặc thử khởi động lại.
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test DisconnectedView render DisconnectedView icon sensor 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5F5",
          "borderColor": "white",
          "borderRadius": 40,
          "borderWidth": 0,
          "height": 80,
          "width": 80,
        },
        undefined,
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F0F2F5",
            "borderColor": "white",
            "borderRadius": 32,
            "borderWidth": 0,
            "height": 64,
            "width": 64,
          },
          undefined,
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <SvgMock
        height={38}
        width={38}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 24,
        "marginTop": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "color": "#F5222D",
            "fontSize": 14,
            "marginLeft": 10,
          },
        ]
      }
    >
      Ngắt kết nối
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 9,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "antoutline",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 12,
              "marginLeft": 10,
            },
          ]
        }
      >
        Gợi ý
        :
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#595959",
              "borderColor": "white",
              "borderRadius": 3,
              "borderWidth": 0,
              "height": 6,
              "width": 6,
            },
            undefined,
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 14,
              "marginLeft": 8,
            },
          ]
        }
      >
        Kiểm tra lại nguồn điện của thiết bị.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#595959",
              "borderColor": "white",
              "borderRadius": 3,
              "borderWidth": 0,
              "height": 6,
              "width": 6,
            },
            undefined,
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 14,
              "marginLeft": 8,
            },
          ]
        }
      >
        Kiểm tra lại Wi-Fi/tín hiệu di động hoặc thử khởi động lại.
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test DisconnectedView render DisconnectedView icon test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5F5",
          "borderColor": "white",
          "borderRadius": 40,
          "borderWidth": 0,
          "height": 80,
          "width": 80,
        },
        undefined,
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F0F2F5",
            "borderColor": "white",
            "borderRadius": 32,
            "borderWidth": 0,
            "height": 64,
            "width": 64,
          },
          undefined,
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <SvgMock
        height={38}
        width={38}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 24,
        "marginTop": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "color": "#F5222D",
            "fontSize": 14,
            "marginLeft": 10,
          },
        ]
      }
    >
      Ngắt kết nối
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 9,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "antoutline",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 12,
              "marginLeft": 10,
            },
          ]
        }
      >
        Gợi ý
        :
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#595959",
              "borderColor": "white",
              "borderRadius": 3,
              "borderWidth": 0,
              "height": 6,
              "width": 6,
            },
            undefined,
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 14,
              "marginLeft": 8,
            },
          ]
        }
      >
        Kiểm tra lại nguồn điện của thiết bị.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#595959",
              "borderColor": "white",
              "borderRadius": 3,
              "borderWidth": 0,
              "height": 6,
              "width": 6,
            },
            undefined,
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 14,
              "marginLeft": 8,
            },
          ]
        }
      >
        Kiểm tra lại Wi-Fi/tín hiệu di động hoặc thử khởi động lại.
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test DisconnectedView render DisconnectedView icon wind 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5F5",
          "borderColor": "white",
          "borderRadius": 40,
          "borderWidth": 0,
          "height": 80,
          "width": 80,
        },
        undefined,
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F0F2F5",
            "borderColor": "white",
            "borderRadius": 32,
            "borderWidth": 0,
            "height": 64,
            "width": 64,
          },
          undefined,
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <SvgMock
        height={38}
        width={38}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 24,
        "marginTop": 8,
      }
    }
  >
    <SvgMock
      height={16}
      width={16}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "color": "#F5222D",
            "fontSize": 14,
            "marginLeft": 10,
          },
        ]
      }
    >
      Ngắt kết nối
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 9,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "antoutline",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 12,
              "marginLeft": 10,
            },
          ]
        }
      >
        Gợi ý
        :
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#595959",
              "borderColor": "white",
              "borderRadius": 3,
              "borderWidth": 0,
              "height": 6,
              "width": 6,
            },
            undefined,
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 14,
              "marginLeft": 8,
            },
          ]
        }
      >
        Kiểm tra lại nguồn điện của thiết bị.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 16,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#595959",
              "borderColor": "white",
              "borderRadius": 3,
              "borderWidth": 0,
              "height": 6,
              "width": 6,
            },
            undefined,
            undefined,
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": undefined,
              "lineHeight": undefined,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {
              "color": "#595959",
              "fontSize": 14,
              "marginLeft": 8,
            },
          ]
        }
      >
        Kiểm tra lại Wi-Fi/tín hiệu di động hoặc thử khởi động lại.
      </Text>
    </View>
  </View>
</View>
`;
